commands/util.go

```diff
@@ -134,45 +142,52 @@ func GetGroupID(owner string) int {
 	return int(gidInt)
 }
 
-// CompressData compresses and base64 encodes a string to place into Consul's KV store.
-func CompressData(data string) string {
+// CompressData compresses a string to place into Consul's KV store.
+func CompressData(data string) ([]byte, uint64) {
 	var compressed bytes.Buffer
-	gz, _ := gzip.NewWriterLevel(&compressed, gzip.BestCompression)
+	gz := brotli.NewWriterLevel(&compressed, brotli.BestCompression)
 	gz.Write([]byte(data))
-	gz.Flush()
 	gz.Close()
-	encoded := base64.StdEncoding.EncodeToString(compressed.Bytes())
+	encoded := compressed.Bytes()
 	Log(fmt.Sprintf("compressing='true' full_size='%d' compressed_size='%d'", len(data), len(encoded)), "info")
-	return encoded
+	return encoded, BROTLI
 }
 
-// DecompressData base64 decodes and decompresses a string taken from Consul's KV store.
-func DecompressData(data string) string {
-	if data != "" {
-		// If it's been compressed, it's been base64 encoded.
-		raw, err := base64.StdEncoding.DecodeString(data)
+// DecompressData decompresses a string taken from Consul's KV store.
+func DecompressData(data []byte, compressFlags uint64) string {
+	if data == nil {
+		return ""
+	}
+
+	var r io.Reader
+	switch compressFlags {
+	case GZIPb64:
```

Comment:
If im reading this right, it seems like things already compressed in consul with gzipb64 will not be decompressed here because they wouldn't have `compressFlags` set yet?

Reply:
They have the `compressFlags` set to 0 due to 0 being the default value for a flag.
Has `GZIPb64` is 0, this match :)